docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+ * A 'variables' partial which defines the colors, typography, grid spacing, animation durations and other foundational elements of the design system
+ * Component partials which define the styles for re-usable components, including the different variations of a component and media queries for responsive layout. These may assume
+   the existence of other variables or mixins but should not import anything themselves.
+
+## Selectors
+
+ * Use simple non-nested class selectors for the majority of styling
+ * Descendant selectors should be avoided. They [are slow](https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Writing_efficient_CSS#Avoid_the_descendant_selector.21) and more importantly, make it easy to create unintended results due to naming clashes or [specificity conflicts](http://css.maxdesign.com.au/selectutorial/advanced_conflict.htm).
+
+## Naming
+
+ * Use [BEM-style](http://csswizardry.com/2013/01/mindbemding-getting-your-head-round-bem-syntax/) naming for components:
+
+ ```js
+ .component { ... }
+ .component__element { ... }
+ .component__element--modifier { ... }
+ ```
+
+ Where 'component' is a re-usable building block (eg. an input field),
+ 'element' is some part of the component which is not reusable on its own, and
+ '--modifier' represents some static variation of a component (eg.
+ `btn--danger` for a button which performs some destructive action)
+ * Use `is-$state` names for state classes (eg. `is-active`) that are dynamically added or removed from JS code
+ * Use names with a `js-` prefix for classes that are used by JS to get a reference to particular DOM elements. These classes should not appear in stylesheets
+ * Except for tests, only styles with `js-` and `is-` prefixes should be referenced by JavaScript code
+ * Use a `u-` prefix for utility classes. eg. A class for hiding DOM elements would be called `u-hidden`
+
+## SASS Features
+
+ * Use variables and mixins
+ * [Avoid @extends](https://www.sitepoint.com/avoid-sass-extend/). Use mixins instead
+ * Use nesting sparingly
+
+## Colors and Typography
+
+ * Components should use colors and font sizes defined in the 'variables' partial (or one of its include) rather than hard-coding colors
+
+## Layout
+
+ * Most layout should be done with block, inline or flexbox layout
+ * Floats should generally be avoided in favor of flexbox. This avoids the need for clearfix and other hacks. Note that flexbox [may not be appropriate](https://hyp.is/AVKcwo8BvTW_3w8LypJ5/jakearchibald.com/2014/dont-use-flexbox-for-page-layout/) for overall page layout
+
+## Vendor Prefixes
+
+ * Our CSS is post-processed with [autoprefixer](https://github.com/postcss/autoprefixer) so vendor prefixes are not needed
+
+## Browser Support
+
+ * Hypothesis projects should be functional on Internet Explorer 10 and above. CSS features that are only available in newer browsers can still be used but the site should behave gracefully in older browsers
```

Comment:
- [ ] Perhaps add a mention for using [@supports](https://developer.mozilla.org/en-US/docs/Web/CSS/@supports) here?